src/client/metadataApiDeploy.ts

```diff
@@ -313,4 +322,31 @@ export class MetadataApiDeploy extends MetadataTransfer<MetadataApiDeployStatus,
   protected async post(result: MetadataApiDeployStatus): Promise<DeployResult> {
     return new DeployResult(result, this.components);
   }
+
+  private async getZipBuffer(): Promise<Buffer> | undefined {
+    if (this.options.mdapiPath) {
+      if (!fs.existsSync(this.options.mdapiPath) || !fs.lstatSync(this.options.mdapiPath).isDirectory()) {
+        throw new Error(`Deploy directory ${this.options.mdapiPath} does not exist or is not a directory`);
+      }
```

Comment:
Should we also ensure a manifest is in the dir?  it would save an http call if it wasn't.  Same for the zipPath option.